### PR TITLE
[NN] Fix: GMMConv to follow formulation

### LINKED ISSUE #4885: GMMConv torch implementation not true to formulation
## 🐛 Bug

In the documentation of [GMMConv](https://docs.dgl.ai/en/0.9.x/generated/dgl.nn.pytorch.conv.GMMConv.html#dgl.nn.pytorch.conv.GMMConv). The kernels are first averaged before aggregating according to the formulation. This means that the messages should not contain a K dimension.

In the code, however, message passing is done before the averaging of the kernels. See the two lines below

https://github.com/dmlc/dgl/blob/ee9887d66a9eb6dc9663a5238e1874b7646acf3a/python/dgl/nn/pytorch/conv/gmmconv.py#L240

Also, there shouldn't be a sum here
https://github.com/dmlc/dgl/blob/ee9887d66a9eb6dc9663a5238e1874b7646acf3a/python/dgl/nn/pytorch/conv/gmmconv.py#L241

Do let me know if this is intended

## To Reproduce

1.
1.
1.

## Expected behavior
The message func should be something like this

```python
def gmm_message(lhs_field, rhs_field, out):
    def edge_fn(edges):
        h = edges.src[lhs_field]
        w = edges.data[rhs_field]

        rst = (h * w).sum(-2)  # or mean if you follow the formulation

        return {out: rst}

    return edge_fn

```

## Environment

 - DGL Version (e.g., 1.0): 0.9.1
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3): Pytorch
 - OS (e.g., Linux):
 - How you installed DGL (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version (if applicable):
 - GPU models and configuration (e.g. V100):
 - Any other relevant information:

### PR DESCRIPTION
**Pull request by lwj5**
## Description

Fix incorrect logic in GMMConv
Fixes #4885 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.*
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage**
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [x] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

*No changes to formatting
**No changes to tests

## Changes

Correct logic to agglomerate the kernels before message passing.

### CODE REVIEW

**Comment by Rhett-Ying** [comment]
@dgl-bot

**Comment by Rhett-Ying** [comment]
@dgl-bot

File: python/dgl/nn/pytorch/conv/gmmconv.py [hunk 1]
```diff
@@ -229,16 +229,18 @@ def forward(self, graph, feat, pseudo):
...
             feat_src, feat_dst = expand_as_pair(feat, graph)
-            graph.srcdata['h'] = self.fc(feat_src).view(-1, self._n_kernels, self._out_feats)
+            graph.srcdata['h'] = self.fc(feat_src).view(-1, self._out_feats, self._n_kernels)
             E = graph.number_of_edges()
             # compute gaussian weight
             gaussian = -0.5 * ((pseudo.view(E, 1, self._dim) -
                                 self.mu.view(1, self._n_kernels, self._dim)) ** 2)
             gaussian = gaussian * (self.inv_sigma.view(1, self._n_kernels, self._dim) ** 2)
-            gaussian = th.exp(gaussian.sum(dim=-1, keepdim=True)) # (E, K, 1)
+            gaussian = th.exp(gaussian.sum(dim=-1, keepdim=False)) # (E, K)
```
[the remark this answers is not in the document]
Comment (lwj5) [reply python/dgl/nn/pytorch/conv/gmmconv.py:238]: This is correct the weight of the gaussian kernel is a scalar value given by 𝑤𝑘(𝑢) here https://docs.dgl.ai/en/0.8.x/generated/dgl.nn.pytorch.conv.GMMConv.html

**Comment by lwj5** [comment]
After consulting with the paper, the current implementation is correct. Instead, it is the formulation provided in the docs here https://docs.dgl.ai/en/0.8.x/generated/dgl.nn.pytorch.conv.GMMConv.html to be incorrect.

The sum of 1 to k should be on the left of the aggregate function, and there's no need for 1/k since it is a division by a constant.

I'll go ahead and close this.

**Pull request closed without merging**